core/notification/src/main/java/edu/stanford/spezi/core/notification/fcm/FCMNotificationService.kt

```diff
@@ -0,0 +1,65 @@
+package edu.stanford.spezi.core.notification.fcm
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import dagger.hilt.android.AndroidEntryPoint
+import edu.stanford.spezi.core.coroutines.di.Dispatching
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.core.notification.notifier.FirebaseMessage
+import edu.stanford.spezi.core.notification.notifier.FirebaseMessage.Companion.ACTION_KEY
+import edu.stanford.spezi.core.notification.notifier.FirebaseMessage.Companion.IS_DISMISSIBLE_KEY
+import edu.stanford.spezi.core.notification.notifier.FirebaseMessage.Companion.MESSAGE_ID_KEY
+import edu.stanford.spezi.core.notification.notifier.NotificationNotifier
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+/*
+ * Service to handle incoming notifications from Firebase Cloud Messaging (FCM)
+ * and register the device with the server when a new token is generated.
+ */
+@AndroidEntryPoint
+class FCMNotificationService @Inject internal constructor(
```

Comment:
This is an android component (android service) and similar to activities we can't initialize or inject directly, but the initialization happens from OS e.g. via `context.startService`, hence `@Inject internal constructor` is not needed

Reply:
now i understand why i couldn't inject anything in the constructor, thanks for the explanation 🚀